api/admin/routes.py

```diff
@@ -729,6 +950,86 @@ def discovery_service_library_registrations():
 @requires_admin
 @requires_csrf_token
 def custom_lists():
+    """Get a full list for the current user's library, or update a specific list
+    ---
+    get:
+      tags:
+        - lists
+      summary: Return a full set of lists defined for the current library
+      description: |
+        Fetches a full set of lists for the library currently set in the Flask request.
+        The current user must have librarian privledges at that library to view the lists.
+        
+        These are structured as objects with an id, name and an array of collections.
+        Each collection within the list is returned as a basic object.
+      security:
+        - BasicAuth
+      parameters:
+        - in: path
+          name: library_short_name
+          description: Short identifying code for a library
+          schema:
+            type: string
+      responses:
+        200:
+          description: Returns a simple object that contains an array of list objects
+          content:
+            application/json:
+              schema: CustomListResponse
+        403:
+          description: The current admin is not an authorized librarian for the current library
+          content:
+            application/json:
+              schema: ProblemResponse 
+    post:
+      tags:
+        - lists
+      summary: Create or Update a custom list for the current library
+      description: |
+        This endpoint accepts a submitted form that can contain JSON data for entries and collections
+
+        If a list already exists with the provided identifier it is updated.
+        If no list exists a new one is created with the relevant details in the current library
+
+        If during this process a work is newly added to this list, associated lanes and counts will also be updated.
+      security:
+        - BasicAuth
+      parameters:
+        - X-CSRF-Token
+        - in: path
+          name: library_short_name
+          description: Short identifying code for a library
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          multipart/form-data:
+            schema: CustomListUpsertPost
+      responses:
+        200:
+          description: A text response containing only the id of the updated list object
+          content:
+            text/html:
+              example: 1234
+        201:
+          description: A text response containing only the id of a newly created list
+          content:
+            text/html:
+              example: 1234
+        4XX:
+          description: |
+            One of the following endpoint-specific errors
+            * `AdminNotAuthorized`: Current admin cannot perform this operation for this library
+            * `MISSING_CUSTOM_LIST`: No list could be found for provided identifier
```

Comment:
I don't know how to address this problem, but these are different types of things. AdminNotAuthorized is an exception class that can be converted into a ProblemDetail object, and MISSING_CUSTOM_LIST is a ProblemDetail constant.

As you note, the thing that actually gets returned is a ProblemResponse containing the information conveyed by a ProblemDetail object. I don't know what to do here since that ProblemDetail could have been one of the constants, derived from one of the constants, instantiated via an exception,  or instantiated some other way.

1. The distinguishing feature of a ProblemDetail is the URI which uniquely identifies the type of problem, so maybe that should be the identifying factor.
2. The ProblemError class (in core/util/problem_detail.py) might be a useful way to think about this. Maybe making it so it can be treated _either_ as a raised exception _or_ a `ProblemResponse` would simplify things.

Reply:
I think using the URI as the identifier for the type of response would be the best bet, since that is more descriptive and unique. I tried, for all of these endpoints, to follow the logic as far as I could in looking what could raise (anticipated) exceptions. But I missed the `ErrorHandler` which catches unanticipated errors. So I think for these codes I would update this section to be:
- 403: An authentication/permission error that returns an HTML page
- 4XX: Anticipated exceptions identified by URI
- 5XX: Unanticipated error that can return a `ProblemResponse`
- 5XX: Unanticipated error due to a bug that returns a text resopnse